Public/Standalone/Set-PDFForm.ps1

```diff
@@ -28,5 +29,11 @@ function Set-PDFForm {
             $FormField.setValue($FieldNameAndValueHashTable[$Key]) | Out-Null
         }
     }
+    
+    if($Flatten)
+    {
+        $PDFAcroForm.flattenFields()
```

Comment:
What happens if there is no forms and you try to flatten it? Is it possible try/catch is needed here?

Reply:
Just tested with a empty pdf, still works with no error.
If I pass the hash table to the empty pdf, we get this in powershell, but the output file is still created.

```
WARNING: No form field with name p.Name found
WARNING: No form field with name p.ARN found
WARNING: No form field with name p.Type found
WARNING: No form field with name p.Date found
```

I guess we could wrap in a catch, but a catch around the call to set-pdfform is probably a better idea. That way the user can do the error handling in the main script.

Reply:
If it fires warning it won't be caught by try/catch as far as a I know. I don't believe error handling should be left to outside. There should be 2 versions - 1 to convert errors to warning and let the script continue, one to throw on error when someone uses ErrorAction Stop. What I am missing in the commandlet is check for existiance of of source (and use of COnvert-Path as shown in other examples). Right now it would error out if given non-existing path or path on network drive or relative path :)

Reply:
Hmm, this is starting to get beyond my knowledge, I am not a dev, just a admin that likes to automate :)

I can wrap the calls to the iText7 functions in try-catch, and probably a good idea if I made the two filepaths mandatory and do a check-path on them. I have no idea how to deal with the ErrorAction thing.. 

How do I edit my proposed code? I cant find a edit button anywhere.

Reply:
You edit it in your fork, it will automatically update the PR as long as it's open